src/main/java/net/datafaker/providers/base/Internet.java

```diff
@@ -66,7 +66,36 @@ public String domainSuffix() {
         return resolve("internet.domain_suffix");
     }
 
+    /**
+     * Returns a web domain.
+     * 
+     * @return a domain in the form "www.example.com"
+     * @deprecated Since 2.0.0 
+     * @see #hostname()
+     * @see #uri()
+     */
+    @Deprecated(since="2.0.0",forRemoval = true) 
     public String url() {
+        return hostname();
+    }
+
+    /**
+     * Returns a string representing a URL, in the form "https://www.example.com/lorem/ipsun".
+     * 
+     * @return a URL, in the form "https://www.example.com/lorem/ipsun"
+     * @since 2.0.0
+     */
+    public String uri() {
+        return "https://" + hostname() + "/" + slug(faker.lorem().words(2), "/");
```

Comment:
since it's only `hhtps` shouldn't it be called httpsurl?
there are could be others e.g. `http`, `https`, `ftp`, `mailto`, `file`, `data` and `irc`

based on https://en.wikipedia.org/wiki/URL.wikipedia.org/wiki/Internet_Relay_Chat#URI_scheme)

Reply:
And `tel` and `ws` and `wss` nearly un-ending….

I guess the question is do we want to maximize technical correctness or usability? Arguably we could also include port randomly. Username and password for http/https/ftp, etc.

I went with what would seem obvious to users (from my perspective). I figured assuming https in 2023 was fine, but I guess I could randomly use with or without the s 🤷‍♂️

Reply:
I wouldn't remove URL. An URL is different from a URI, would be good to keep them both.

Reply:
It wasn’t removed. I agree they’re different things. I’ll drop the scheme to make the uri method more correct, and randomly include creds, port, params, and anchor as well as renaming it to weburi I guess. Or maybe I do include scheme randomly as well and call it weburl 🤷‍♂️

Reply:
i have an idea how it could be done (may be in a separate PR)
1. there could be an enum with all supported urls e.g.
```java
enum UrlType {
  HTTP('http', new HttpUrlGenerator()),
  HTTPS('https', new HttpsUrlGenerator()),,
  FTP('ftp', new FtpUrlGenerator()),,
  ...
}
```
for which there could be some rules specified.
then there could be methods
`url(UrlType type)` which generates based on requested type
and `url()` which is same as `url(Option.option(UrlTyps.class))` which means that first it picks a random type and then runs url generation for that type

Reply:
I do like that idea, but I think it's more effort than anticipated for this PR :wink:

Reply:
that's why i mentioned that (may be in a separate PR)

Reply:
Yup, I don't seem to be getting myself across clearly today. I just wanted you to know I was in agreement, but that I wouldn't be tackling it here. (I appreciate that you suggested or acknowledged it up front.)

Reply:
:) @kingthorin you were both clear, and you were both in agreement: simple change now, maybe more extended change later.

Reply:
Thanks btw for looking into this @kingthorin , nice work!